llvm/lib/Target/SPIRV/Analysis/SPIRVConvergenceRegionAnalysis.cpp

```diff
@@ -56,20 +56,8 @@ getConvergenceTokenInternal(BasicBlockType *BB) {
       "Output type must be an intrinsic instruction.");
 
   for (auto &I : *BB) {
-    if (auto *II = dyn_cast<IntrinsicInst>(&I)) {
-      switch (II->getIntrinsicID()) {
-      case Intrinsic::experimental_convergence_entry:
-      case Intrinsic::experimental_convergence_loop:
-        return II;
-      case Intrinsic::experimental_convergence_anchor: {
-        auto Bundle = II->getOperandBundle(LLVMContext::OB_convergencectrl);
-        assert(Bundle->Inputs.size() == 1 &&
```

Comment:
This assert is wrong. The only valid assert here is to say ``assert(!Bundle)``, because the bundle cannot exist. At that point, the only thing that can be returned is ``II` itself, because ``TII`` cannot exist.

Reply:
Agreed this assert is wrong, but the question I have is if the analysis will do the correct thing if `II` is returned. I'm guessing we do not test that.